core/tests/test_utils.py

```diff
@@ -2242,6 +2242,15 @@ def get_json(self, url, params=None, expected_status_int=200):
 
         return self._parse_json_response(json_response, expect_errors)
 
+    def post_req(self, url, expected_status_int=200):
```

Comment:
Using this function as the post_json usage in the above tests gives the following error
```
Traceback (most recent call last):
  File "/home/farees/opensource/oppia/core/controllers/android_e2e_config_test.py", line 104, in test_check_if_topic_exists_and_not_published
    '/initialize_android_test_data', payload={}, csrf_token=csrf_token)
  File "/home/farees/opensource/oppia/core/tests/test_utils.py", line 2277, in post_json
    return self._parse_json_response(json_response, expect_errors)
  File "/home/farees/opensource/oppia/core/tests/test_utils.py", line 2218, in _parse_json_response
    self.assertEqual(json_response.content_type, 'application/json')
AssertionError: 'text/html' != u'application/json'
```

not sure why it gives 'text/html' content-type for a post request.
also avoiding the csrf_token which is required in the post_json 

Created this function showing how to make the tests to do the desired assertions (will remove this once I get to know the preferable approach)

Reply:
It seems like ``post_json`` might be wrong. It's calling ``_send_post_request`` which calls TestApp's ``post`` method, but per https://docs.pylonsproject.org/projects/webtest/en/2.0.4/testapp.html#making-json-requests we should be using ``post_json`` for JSON messages to ensure the correct type is being set. I'm surprised we haven't hit this before.

@nithusha21 do you have any thoughts on this? We ought to not have a second method here, but changing ``post_json`` in this PR may have unexpected side effects codebase-wide.

Reply:
@FareesHussain If you take a look at the impl of post_json, it seems like there's an "if csrf_token" clause. I wonder if you can just pass None in and update the docstring accordingly -- would that work?

For json vs post_json, it's probably fine to change post to post_json if needed, as long as the backend tests still pass after the change. We haven't hit this error before though so I'm wondering whether there's something different about Farees's new code additions rather than there being an error in test_utils.

Reply:
@seanlip there is one more issue here, 
```
Traceback (most recent call last):
  File "/home/farees/opensource/oppia/core/controllers/android_e2e_config_test.py", line 104, in test_check_if_topic_exists_and_not_published
    '/initialize_android_test_data', payload={}, csrf_token=csrf_token)
  File "/home/farees/opensource/oppia/core/tests/test_utils.py", line 2277, in post_json
    return self._parse_json_response(json_response, expect_errors)
  File "/home/farees/opensource/oppia/core/tests/test_utils.py", line 2218, in _parse_json_response
    self.assertEqual(json_response.content_type, 'application/json')
AssertionError: 'text/html' != u'application/json'

```

`_parse_json_response` gives the above error in the tests.

So I decided to add another parameter `return_json` which is true by default and will be false in this condition.
Is this is a good approach?

Reply:
Probably not. What's different about your new controller that causes this error to appear here and not previously?

Reply:
Done, I forgot using self._render_json() here